src/ui/state/stakeholder.rs

```diff
@@ -52,15 +49,15 @@ impl StakeholderHomeState {
         }
     }
 
-    fn update_vaults(&mut self, vaults: Vec<(Vault, VaultTransactions)>) {
+    fn update_vaults(&mut self, vaults: Vec<Vault>) {
```

Comment:
Looks like it does not need to take a Vec, as self.calculate_balance() already takes a slice :)

Reply:
Yes, but I will let it like that because we will need the vec later